File: features/merchant/reputation/src/main/java/com/tokopedia/tkpd/tkpdreputation/domain/interactor/GetLikeDislikeReviewUseCase.java
package com.tokopedia.tkpd.tkpdreputation.domain.interactor;

import com.tokopedia.core.base.domain.RequestParams;
import com.tokopedia.core.base.domain.UseCase;
import com.tokopedia.core.base.domain.executor.PostExecutionThread;
import com.tokopedia.core.base.domain.executor.ThreadExecutor;
import com.tokopedia.tkpd.tkpdreputation.domain.model.GetLikeDislikeReviewDomain;
import com.tokopedia.tkpd.tkpdreputation.inbox.data.repository.ReputationRepository;

import rx.Observable;

/**
 * @author by nisie on 9/29/17.
 */

public class GetLikeDislikeReviewUseCase extends UseCase<GetLikeDislikeReviewDomain> {

    private static final String PARAM_REVIEW_IDS = "review_ids";
    private static final String PARAM_USER_ID = "user_id";

    ReputationRepository reputationRepository;


    public GetLikeDislikeReviewUseCase(ThreadExecutor threadExecutor,
                                       PostExecutionThread postExecutionThread,
                                       ReputationRepository reputationRepository) {
        super(threadExecutor, postExecutionThread);
        this.reputationRepository = reputationRepository;
    }

    @Override
    public Observable<GetLikeDislikeReviewDomain> createObservable(RequestParams requestParams) {
        return reputationRepository.getLikeDislikeReview(requestParams);
    }

    public static RequestParams getParam(String reviewIds, String userId) {
        RequestParams params = RequestParams.create();
        params.putString(PARAM_REVIEW_IDS, reviewIds);
        params.putString(PARAM_USER_ID, userId);
        return params;
    }
}

File: features/merchant/shop_page/src/main/java/com/tokopedia/shop/review/shop/domain/GetLikeDislikeReviewUseCase.kt
package com.tokopedia.shop.review.shop.domain

import com.tokopedia.shop.review.shop.domain.model.GetLikeDislikeReviewDomain
import com.tokopedia.shop.review.shop.domain.repository.ReputationRepository
import com.tokopedia.usecase.RequestParams
import com.tokopedia.usecase.UseCase
import rx.Observable

/**
 * @author by nisie on 9/29/17.
 */
class GetLikeDislikeReviewUseCase(private val reputationRepository: ReputationRepository) : UseCase<GetLikeDislikeReviewDomain>() {
    override fun createObservable(requestParams: RequestParams): Observable<GetLikeDislikeReviewDomain> {
        return reputationRepository.getLikeDislikeReview(requestParams)
    }

    companion object {
        private const val PARAM_REVIEW_IDS = "review_ids"
        private const val PARAM_USER_ID = "user_id"
        fun getParam(reviewIds: String?, userId: String?): RequestParams {
            val params = RequestParams.create()
            params.putString(PARAM_REVIEW_IDS, reviewIds)
            params.putString(PARAM_USER_ID, userId)
            return params
        }
    }

}